DataDog/datadog-agent
# [System Probe] Zip system probe /debug/stats

### What does this PR do?

This adds a function which zips System Probe stats into the expvar directory when a flare is made. 

### Motivation

Better troubleshooting

### Additional Notes

This utilizes the RemoteSysProbeUtil in the status package, which might see some changes to be better supported by importing the sysprobe config in the main system binary. 

### QA Steps:

- Install agent from release candidate branch
- Enable processes in datadog.yaml, and system-probe in system-probe.yaml
- Run Agent
- Execute agent flare (command varies based upon OS)
 - https://docs.datadoghq.com/agent/basic_agent_usage/
- Do not send flare, have it written locally
- Unpack flare archive
- Check for file in the archive /<flare_archive_folder>/expvar/system-probe
- On supported OS's, this should include all the stats you can get from the system probe by running:
 - curl localhost/debug/stats --unix-socket /opt/datadog-agent/run/sysprobe.sock | jq .
- On unsupported OS's, you should see a message:

```
"System Probe is not supported on systems not running the process agent"
```

Supported OS's are those that have the system probe listening on the socket location:

-  /opt/datadog-agent/run/sysprobe.sock
